api/validation/etcd.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"strings"
+
+	"github.com/gardener/etcd-druid/api/v1alpha1"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateEtcd validates Etcd object.
+func ValidateEtcd(etcd *v1alpha1.Etcd) field.ErrorList {
+	return validateEtcdSpec(etcd, field.NewPath("spec"))
```

Comment:
Validating resources should always validate the object meta, see e.g. https://github.com/gardener/gardener/blob/0212d5b5d90674b693e6110420b189a308ecc288/pkg/apis/extensions/validation/backupentry.go#L27-L31.

```suggestion
	allErrs := field.ErrorList{}
	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&etcd.ObjectMeta, false, apivalidation.NameIsDNSSubdomain, field.NewPath("metadata"))...)
	allErrs = append(allErrs, validateEtcdSpec(etcd, field.NewPath("spec"))...)
	return allErrs
```